Restructure TOC and add get started (#276)
grafana/writers-toolkit: docs/sources/get-started/_index.md

---
description: Get started with the Writer's Toolkit
menuTitle: Get started
title: Get started with the Writer's Toolkit
weight: 50
aliases:
  - /docs/writers-toolkit/get-started/
---

# Get started with the Writer's Toolkit

This guide provides includes everything you need to complete your documentation project from start to finish.

> In short, it's a `tl;dr` on Writer’s Toolkit to quickly point you in the right direction.

Reviewer: I would consider `tl:dr` to be jargon.  I'm not sure that it's globally understood (and even if it is, always spell out acronyms on first use).

| ---------------------------------- | ---------------------------------------------------------------------- |
| Add docs to your GitHub repository | Add a `docs/sources` folder to your repo, if it doesn’t already exist. |
| Create the folders and structure   | Use the content framework for consistency across product docs.         |

### 3. Draft the docs

| Action                                         | Considerations                                                                                                                                                                                   |
| ---------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| Learn how to best write and organize your docs | Read the [topic types]({{< relref "../structure/topic-types" >}}) topic and sub-topics in the Writers’ Toolkit.                                                                                  |

> | Add front matter                               | [Front matter]({{< relref "../write/front-matter" >}})der, and other key information to Hugo (our publishing platform). Learn more about [Front matter]({{< relref "../write/front-matter" >}}). |

Reviewer: ```suggestion
| Add front matter                               | [Front matter]({{< relref "../write/front-matter" >}}), and other key information to Hugo (our publishing platform). Learn more about [Front matter]({{< relref "../write/front-matter" >}}). |
```

Author: I'm making this change in the file directly (since I have it open to address some table-related issues).

---

> description: Get started with the Writer's Toolkit

Reviewer 2: ```suggestion
description: Get started with Writers’ Toolkit
```
The title is considered a proper noun so the article is extraneous, and was in the plural possessive. I am personally fine with singular possessive.

Reviewer 2: Please change this throughout for consistency.

Author: Will do. Thank you for the suggestion.

Reviewer: Sorry to disagree that Writer's Toolkit is a proper noun. If I saw a reference to the Writer's Toolkit without the article I would put it back in.  This may be a situation where the rules of usage don't 100% follow the correct rules of grammar, but just a bare "Writer's Toolkit" without an article sounds wrong to my ear.

Author: I changed it to Writers' Toolkit throughout the content. I agree that it sounds weird without 'the' article. Also, plural posessive seems weird since I've always written it as singular possessive. At least it's consistent now.

Reviewer 2: > Unfortunately, the original decision wasn't discussed in a public channel but in a multi-person DM that included Brenda, Chris, Eve, and Fiona. I believe it was Ursula who originally proposed it and ensured that the initial publication of Writers' Toolkit had consistent usage.
> 
> In that conversation it was decided that Writers' Toolkit is the name of the project, to quote:
> 
> > Writers' Toolkit - proper noun as it is really very much our thing
> 
> In this sense, it is similar to Grafana Agent or Grafana Mimir.
> 
> I believe it is plural possessive because it is the toolkit for all (many) writers and not just a single writer.
> 
> Writers' Toolkit is definitely our current style but we can open up a roundtable discussion or GitHub issue if we want to discuss that.
> 
> I'm just glad it is no longer the mouthful of "Grafana Technical Writing Contributor's Toolkit" :D

Yes, and my initial input matched @JStickler’s sentiment, where I proposed _Writer’s toolkit_ to match our style and not make it a proper noun. In that case, it would have been "... the _Writer’s toolkit_". But because it was deemed a proper noun, it therefore must not have an article.

Reviewer 2: This is an example of where I agreed to disagree and moved on. Sharing this history here for context about the decision.

menuTitle: Get started
title: Get started with Writers' Toolkit
weight: 50
aliases:
  - /docs/writers-toolkit/get-started/
---

# Get started with Writers' Toolkit

This guide provides includes everything you need to complete your documentation project from start to finish.

> In short, it quickly points you in the right direction.

Reviewer: Re-reading this, it occurs to me that while we mention it elsewhere in Writer's Toolkit, this would be a good place to add just a sentence or two that mentions markdown and Hugo.  Then line 46 (add `docs/sources`) makes more sense.

## Goals

By reading this guide, you will:

- Learn how to successfully document new products or features
- Have a step-by-step guide from project kick-off to release
- Be introduced to the craft of creating good documentation

## Audience

- Software developers who need to create docs alongside code
- Other technical roles that create and contribute to docs

## Before you begin

- Consider what you want someone reading your documentation to understand or accomplish

## Project checklist

### 1. Plan the work

| Action | Considerations |
|---|---|

> | Decide what kind of documentation deliverables you need | What do you want a user to accomplish? |

Author: I'm not sure I understand the purpose of adding all of this reformatting. It's making the table hard to read for a users. Is there a layout benefit?

> | Decide what kind of documentation deliverables you need | What do you want a user to accomplish? |

Reviewer 3: Prettier insists that tables have aligned columns in the source. In general, that makes the tables more easy to read than misaligned columns even if they are semantically equivalent.

> | Decide what kind of documentation deliverables you need | What do you want a user to accomplish? |

Author: Okay. What prettier insists on, then I'll approve it. Thank you for the explanation.

| Know which release life cycle stage the product or feature is at | Review the [Release cycle for Grafana Labs](/docs/release-life-cycle/) <br /> Check in at the [Grafana Community Slack #docs channel](https://grafana.slack.com/archives/CNCRV74GP) to learn about docs for each of these stages. |
| Consider content reuse, if applicable | Read the [Reuse strategy documentation]({{< relref "../write/reuse-content/reuse-directories/" >}}) <br />Check in at #docs to find out how the docs team can support you in this. |

### 2. Create the structure

| Action | Considerations |
|---|---|
| Add docs to your GitHub repository | Add a `docs/sources` folder to your repo, if it doesn’t already exist. |
| Create the folders and structure | Use the content framework for consistency across product docs. |

### 3. Draft the docs

| Action | Considerations |
|---|---|

> | Learn how to best write and organize your docs | Read the [topic types](https://grafana.com/docs/writers-toolkit/writing-guide/topic-types/) topic and sub-topics in the Writers’ Toolkit. |

Author: ```suggestion
| Learn how to best write and organize your docs | Read the [topic types]({{< relref "../write/topic-types/" >}}) topic and sub-topics in the Writers’ Toolkit. |
```

| Add front matter | Front matter provides metadata, page title, display order, and other key information to Hugo (our publishing platform). Learn more about [Front matter]({{< relref "../write/front-matter/" >}}).  |
| Add images and media  | [Images and media]({{< relref "../write/image-guidelines/" >}})  |
| Use links and references  | [Links and references]({{< relref "../write/references/" >}})   |

> | Add code samples   | [Code samples]({{< relref "../write/markdown-guide/#code-blocks" >}})|

Author: This suggestion would break the link. 
```suggestion
| Add code samples   | [Code samples]({{< relref "../write/markdown-guide#code-blocks" >}})|
```

### 5. Publish the docs

{{% responsive-table %}}

| Action                                       | Considerations                                                                                                                                                                    |
| -------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Review publishing options                    | Engage #docs to discuss the publishing options for different stages in the release cycle.                                                                                         |
| Backport, if required                        | See the [backporting guidelines]({{< relref "../review/backporting" >}}).                                                                                                         |
| Add What’s New or release notes, if required | See the [Contribute to release notes guidelines]({{< relref "../contribute-documentation/contribute-release-notes" >}}). \*In Grafana, use the label add-to-whats-new in your PR. |

> {{% /responsive-table %}}

Author: @jdbaldry This one change from prettier is preventing the PR from passing. I can't commit the change because there's nothing in the code that is changed. Running prettier PR locally on this branch returns no changes. Any suggestions?

> {{% /responsive-table %}}

Reviewer 3: I bet that it is a missing trailing newline at the end of the file.

> {{% /responsive-table %}}

Reviewer 3: I think the suggestion is improperly provided. I'll fix it up to unblock the PR.
